### ISSUE TITLE
[moor_ffi] Adding result code to SqliteException

### ISSUE THREAD

**Issue by alextekartik** [simolus3/moor#509]
Thanks for you moor_ffi that I use to provide a sqflite ffi (https://pub.dev/packages/sqflite_common_ffi) implementation to allow unit test on the desktop. It is very well done. I was even thinking that it could make a separate package outside of moor, but well, it works very well the way it is.

I have a small suggestion. Some sqflite users relies on some internal to get the SQLite result code on iOS and Android (sometimes extracted with some ugly text parsing).

In moor_ffi, the error code is plain text such as `near "DUMMY": syntax error`. However you have sometimes the result code just before creating the SqliteException:

```
if (result != Errors.SQLITE_OK) {
  throw SqliteException(errorMsg);
}
```

or using `SqliteException._fromErrorCode`

Would it be possible to add `resultCode` (when available) to your SqliteException so that it could be use by people who needs it for fine exception handling.

I could work on a PR if you're ok with that.

Thanks!

**Comment by simolus3** [comment]
> Thanks for you moor_ffi that I use to provide a sqflite ffi

Oh cool, I didn't know about that. It looks great!

> I was even thinking that it could make a separate package outside of moor

I also think that this would be a cleaner approach. The main reason that `moor_ffi` contains both the lower-level bindings and moor apis is that I didn't want the overhead of maintaining yet another package. Now that `dart:ffi` is getting more stable it might make sense to move the moor part in `moor_ffi` into the main `moor` package and provide sqlite bindings as an unrelated package.

> add resultCode (when available) to your SqliteException

I think this is reasonable, yes. 

> I could work on a PR if you're ok with that.

That would be great! Again, let me know if you need anything.

**Comment by simolus3** [comment]
@alextekartik Thanks again for your contributions. Do you want this to be released quickly? I can publish it now if you want me to, otherwise I'd wait and release it with the next version of moor (will probably still take a week or two).

**Comment by alextekartik** [comment]
Basically I will implement the same support for resultCode in sqflite. No urgency thanks. Let me know if you publish a dev release before. Thanks!

**Comment by simolus3** [comment]
@alextekartik I've released `moor_ffi` version 0.5.0 which contains the extended result code.

**Comment by alextekartik** [comment]
Thanks!
